packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -115,11 +116,18 @@ export default class RouterService extends Service {
        attempted transition
      @public
    */
-  transitionTo(...args: (string | object)[]) {
+  transitionTo(
+    routeNameOrURL: string,
+    ...modelsAndOptions: [...any[], Record<string, unknown>]
+  ): Transition;
+  transitionTo(routeNameOrURL: string, ...models: any[]): Transition;
+  transitionTo(...modelsAndOptions: [...any[], Record<string, unknown>]): Transition;
+  transitionTo(...models: any[]): Transition;
+  transitionTo(...args: (string | any | Record<string, unknown>)[]): Transition {
     if (resemblesURL(args[0])) {
       // NOTE: this `args[0] as string` cast is safe and TS correctly infers it
       // in 3.6+, so it can be removed when TS is upgraded.
```

Comment:
Here and on lines 340-342 below, the comments are now out of date. Maybe a fast follow so we can actually land this thing, though.

Reply:
I'll correct this but wait to push.

Reply:
Nevermind, I've corrected and pushed.